src/point_one/fusion_engine/messages/fault_control.h

```diff
@@ -0,0 +1,195 @@
+/**************************************************************************/ /**
+ * @brief System fault control messages.
+ * @file
+ ******************************************************************************/
+
+#pragma once
+
+#include "point_one/fusion_engine/messages/defs.h"
+
+namespace point_one {
+namespace fusion_engine {
+namespace messages {
+
+// Enforce 4-byte alignment and packing of all data structures and values.
+// Floating point values are aligned on platforms that require it. This is done
+// with a combination of setting struct attributes, and manual alignment
+// within the definitions.
+#pragma pack(push, 1)
+
+/**
+ * @defgroup fault_control_messages System Fault Control
+ * @brief Messages/types for controlling or simulating system faults.
+ * @ingroup config_and_ctrl_messages
+ */
+
+/**
+ * @brief Available fault types/control inputs.
+ * @ingroup fault_control_messages
+ *
+ * See @ref FaultControlMessage.
+ */
+enum class FaultType : uint8_t {
+  /**
+   * Take no action.
+   *
+   * Payload format: none
+   */
+  NONE = 0,
+  /**
+   * Force the device to crash (intended for factory test purposes only).
+   *
+   * @warning
+   * The device will crash immediately after receiving this request. It will not
+   * send a @ref CommandResponseMessage back to the user.
+   *
+   * Payload format: none
+   */
+  CRASH = 1,
+  /**
+   * Force the device to exhibit a fatal error (intended for factory test
+   * purposes only).
+   *
+   * Unlike @ref FaultType::CRASH, a fatal error will send an error notification
+   * to the user, but will still not send a @ref CommandResponseMessage.
+   *
+   * Payload format: none
+   */
+  FATAL_ERROR = 2,
```

Comment:
For both `FATAL_ERROR` and `CRASH` is it worth going into a bit more detail on the expected behavior of the device along the lines of "Will no longer output position data. If watchdog is enabled the device will reset."

Reply:
Yeah, good idea. I'll add more details to all of them.